### ISSUE TITLE
You can generate TOML that will then fail to load by serializing a large u64

### ISSUE THREAD

**Issue by pedrocr** [alexcrichton/toml-rs#227]
Apparently `u64` gets serialized as if it was an `i64` in the TOML format but instead of failing to serialize on very large `u64` values the library will generate the TOML and then fail to parse it. Here's an example:

```rust
#[macro_use] extern crate serde;
extern crate toml;

#[derive(Serialize, Deserialize)]
struct MyStruct {
    num: u64,
}

fn main() {
    let s = MyStruct{num: 11077308757146580733};
    let toml = toml::to_string(&s).unwrap();
    eprintln!("string is {}", &toml);
    let s: MyStruct = toml::from_str(&toml).unwrap();
    eprintln!("second number is {}", s.num);
}
```

If indeed very large u64 numbers are not serializable as TOML then it should fail on `to_string()` as well as on `from_str()`.

**Comment by alexcrichton** [comment]
Thanks for the report! Unfortunately this is due to how internally toml-rs represents integers as `i64` and 11077308757146580733 is out of bounds for an `i64`. This'd probably require some internal refactoring to work, but seems like it's definitely a bug to fix!

**Comment by pedrocr** [comment]
Yeah, I figured out it was because of `i64`/`u64` and refactored the app appropriately but it's a bit of a landmine because the write succeeds and the read fails. I triggered this when I generated random ID's for peers in `syncer` and suddenly got a heisenbug where some runs would work fine and some would fail. I ended up just switching the random id's to `i64` which matches the sqlite integer type anyway.

Making sure the write fails as well would be a good first step and maybe that's easy to do by just adding that extra check. It still makes it easy for someone to write something that seems to work fine and later fails when the `u64` reaches a number bigger than `i64::MAX`. No idea how that could be fixed.
